internal/contextualtuples/contextual_tuples.go

```diff
@@ -0,0 +1,66 @@
+package contextualtuples
```

Comment:
I'm not sure these changes warrant a whole package specific to contextual tuples. Since contextual tuples are just an extra set of `[]*openfgapb.TupleKey` provided at the request, I'd prefer that we continue to use the `validation` package to validate the tuples (like we've been doing everywhere else).

May I suggest instead of these changes that we simplify a bit to something like:

```
package validation

func ValidateTuples(typesys *typesystem.TypeSystem, tuples ...*openfgapb.TupleKey) error {
    for _, tuple := range tuples {
        if err := ValidateTuple(typesys, tuple); err != nil {
            return err
        }
    }

    return nil
}
```

We already have [`ValidateTuple`](https://github.com/openfga/openfga/blob/main/internal/validation/validation.go#L15). This approach would just extend this existing function to provide a way to validate more than one tuple at a time.

Reply:
> And, regarding the read methods, those could be enforced by passing the ContextualTuples iterator into the filtered iterator so any invalid tuples in the ContextualTuples set are filtered out just like the database iterator.

I just moved stuff around in this PR, this feels like scope creep to me 😬 Can we do this on a follow-up PR after we release today?

Reply:
> We're not only moving packages/code we're also fixing an issue (contextual tuple validation).

I moved stuff around because the existing code validating contextual tuples in Check needed to be used also in ListObjects. So i put them in one place that could be used by both. So this is to say, it's the smallest change I thought of that could make the fix work. If I put the `New` method inside the `validation` package, where would I put the `Read` methods? They don't do validations

Reply:
We already have storage interfaces that deal with filtering reads, and so moving this to a separate package splits the concerns. A single responsibility separation of concerns is a better approach here. Let the storage interface (the filtered tuple iterator) deal with the filtering of the reads not another `contexualtuples` package.

The validation can be re-used across Checks and ListObjects if it's shared in the `validate` package, which is shared across those methods already as well.

Reply:
Agreed offline to send another PR with this change